website/docs/r/elasticsearch_domain.html.markdown

```diff
@@ -108,6 +108,101 @@ resource "aws_elasticsearch_domain" "example" {
   }
 }
 ```
+### VPC based ES
+
+```hcl
+variable "vpc" {}
+
+variable "domain" {
+  default = "tf-test"
+}
+
+data "aws_vpc" "selected" {
+  tags {
+    Name = "${var.vpc}"
+  }
+}
+
+data "aws_subnet_ids" "selected" {
+  vpc_id = "${data.aws_vpc.selected.id}"
+
+  tags {
+    Tier = "private"
+  }
+}
+
+data "aws_region" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_security_group" "es" {
+  name        = "${var.vpc}-elasticsearch-${var.domain}"
+  description = "Managed by Terraform"
+  vpc_id      = "${data.aws_vpc.selected.id}"
+
+  ingress {
+    from_port = 443
+    to_port   = 443
+    protocol  = "tcp"
+
+    cidr_blocks = [
+      "${data.aws_vpc.selected.cidr_blocks}",
+    ]
+  }
+}
+
+resource "aws_iam_service_linked_role" "es" {
+  aws_service_name = "elasticsearch.amazonaws.com"
```

Comment:
i'm not sure this is correct? i get this error: 

```* aws_iam_service_linked_role.es: Error creating service-linked role with name elasticsearch.amazonaws.com: AccessDenied: elasticsearch.amazonaws.com is not a valid AWS service name (Service Principal)
	status code: 403```

the docs here say it should be `es.amazonaws.com` ?